### ISSUE TITLE
Problem with adding page

### ISSUE THREAD

**Issue by oleaha** [divio/django-cms#4578]
Hello!

I think i found a problem with adding and deleting pages. Used django-cms 3.1.2 and upgraded to 3.1.3 since this problem was a fix in this update.

Here is how to reproduce it:

1. Add a new page that is at least one level down from root with the following details: title: Test, slug: test
2. Publish this page
3. Delete the page
4. Now add a page at the same level with the same details. 
5. When you now try to publish this new page you get a 500 error with the following stacktrace

```
[13/Oct/2015 09:15:49] ERROR [django.request:231] Internal Server Error: /admin/cms/page/180/nb-no/publish/
Traceback (most recent call last):
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/django/core/handlers/base.py", line 111, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/django/utils/decorators.py", line 105, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/django/views/decorators/cache.py", line 52, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/django/contrib/admin/sites.py", line 206, in inner
    return view(request, *args, **kwargs)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/django/utils/decorators.py", line 29, in _wrapper
    return bound_func(*args, **kwargs)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/django/views/decorators/http.py", line 41, in inner
    return func(request, *args, **kwargs)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/django/utils/decorators.py", line 25, in bound_func
    return func.__get__(self, type(self))(*args2, **kwargs2)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/django/db/transaction.py", line 394, in inner
    return func(*args, **kwargs)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/reversion/revisions.py", line 297, in do_revision_context
    return func(*args, **kwargs)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/cms/admin/pageadmin.py", line 967, in publish_page
    published = page.publish(language)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/cms/models/pagemodel.py", line 603, in publish
    self._publisher_save_public(public_page)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/cms/models/pagemodel.py", line 1210, in _publisher_save_public
    obj = obj.move(target=public_parent, pos='first-child')
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/cms/models/pagemodel.py", line 1237, in move
    super(Page, self).move(target, pos)
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/treebeard/mp_tree.py", line 1037, in move
    return MP_MoveHandler(self, target, pos).process()
  File "/home/jrc/webapps/orientering/lib/python2.7/site-packages/treebeard/mp_tree.py", line 444, in process
    if self.target.is_descendant_of(self.node):
AttributeError: 'NoneType' object has no attribute 'is_descendant_of'
```

Could this have something to do with recovery of deleted pages and some kind of conflict there?

**Comment by yakky** [comment]
Same as #4576

**Comment by yakky** [comment]
@oleaha would you try https://github.com/yakky/django-cms/tree/feature/treebeard?

**Comment by oleaha** [comment]
Tested the solution, it works fine in our project! @yakky

**Comment by yakky** [comment]
@oleaha great to hear!

**Comment by yakky** [comment]
Closed by #4717
